bento_lib/search/data_structure.py

```diff
@@ -434,6 +434,7 @@ def _resolve(resolve: Tuple[q.Literal, ...], resolving_ds: QueryableStructure, _
     q.FUNCTION_GE: _binary_op(ge),
 
     q.FUNCTION_CO: _binary_op(contains),
+    q.FUNCTION_ICO: _binary_op(contains),
```

Comment:
This will fail with data-structure (non-postgres searches), since contains is a case sensitive operator in python

Reply:
This may be the issue for the other test too
Basically, there are two separate ways these queries can be evaluated - in Postgres, or against a Python object
Eventually we can probably do everything in postgres but for now this is unfortunately the way it's implemented

Reply:
> This will fail with data-structure (non-postgres searches), since contains is a case sensitive operator in python
I was thinking that this function `FUNCTION_ICO` then should be adapted in each sql specific query module (e.g postgres.py). For example, in postgres.py we refine it using `ILIKE` because ILIKE is used in Postgres for case insensitive search.
But in other sql databases, it might be something else and the query will look differently.

Reply:
the data_structure file allows searching on python dictionaries, so no SQL is necessarily involved - this is how we search variants and stuff

Reply:
fixed with #52